compiler/src/main/scala/org/scalajs/core/compiler/PrepJSExports.scala

```diff
@@ -466,6 +475,13 @@ trait PrepJSExports[G <: Global with Singleton] { this: PrepJSInterop[G] =>
           ignoreInvalid = false)
     }
 
+    if (!unitAnnots.exists(_.symbol == JSExportAllAnnotation)) {
```

Comment:
So this means that the following would not warn, right:

```scala
@JSExportAll
class Foo {
  @JSExport("b")
  @JSExport("b")
  def a = ???
}
```

Consider simply dropping this extra case and live with the less good warning. Alternatively bring in the heavy diagnostic:

```
Found duplicate export for name "a". Exports originating from:
    @JSExport("a")
     ^
  @JSExportAll
   ^
```

(you might want to get inspiration from duplicate member definition error messages, etc.)

Reply:
Simplified in https://github.com/scala-js/scala-js/pull/3633/commits/b0f391d38547cea9fc4372e11767a48dbfd15275